subl.fish

```diff
@@ -1,9 +1,37 @@
 # SYNOPSIS
-#   subl [file]
+#   subl [options] [file ...]        edit the given files
+#   subl [options] [directory ...]   open the given directories
+#   subl [options] -                 edit stdin
 #
 
 function subl -d "Open Sublime Text"
-  if test -d "/Applications/Sublime Text.app"
+  set -l opts
+  set -l files
+  set -l projects
+
+  # we'll just ignore anything that looks like an option string...
+  for file in $argv
+    switch $file
+      case '-*'
+        set opts $opts $file
+
+      case '.*' '*'
+        set files $files $file
+    end
+  end
+
+  # if there's one (and only one) *.sublime-project file in the folder listed, open that instead.
+  if [ (count $files) -eq 1 ]
+    set projects (find $files[1] -name "*.sublime-project" -maxdepth 1 2> /dev/null)
+  end
+
+  if [ (count $projects) -eq 1 ]
+    set argv $opts $projects[1]
+  end
+
+  if begin; which subl > /dev/null; and test -x (which subl); end
```

Comment:
first, it's not likely that someone has the sublime plugin installed but doesn't have sublime installed.

second, the shenanigans above are relatively cheap. cheaper than the shenanigans below, trying to figure out if there's a sublime install to call.

third, moving this check above complicates the logic a bit, and the ones that need `eval` have escaping issues thanks to the space in the app bundle name.

fourth, in the event that you've installed the sublime plugin, and called `subl` explicitly, but don't have sublime installed, the extra 50ms or whatever it wastes parsing args is probably the least of your worries :)